coalib/settings/Setting.py

```diff
@@ -223,6 +224,9 @@ def __path__(self, origin=None, glob_escape_origin=False):
         if hasattr(self, 'origin') and self.origin != '':
```

Comment:
when we use a @property, `hasattr(self, 'origin')` wont work.

It was already broken code, because `__init__` sets `self.origin` , so this `hasattr` would never be `False`.

Anyways, you should be able to simply remove it, and then you also dont need the `isinstance(origin, SourcePosition)` below because `self.origin` always returns a string, the filename, which is what we want.

Reply:
If I remove `hasattr(self, 'origin')` then some tests are breaking. It is because `origin` is string in some cases when it is called by `__glob__`.
So i guess we should not remove it.

Reply:
ok, add a comment about it in the code here.